From 45445ba5bd96b7dc9201e4c70b919e9dc98de0aa Mon Sep 17 00:00:00 2001
From: Girish Kumar M P <quic_gmp@quicinc.com>
Date: Sun, 13 Oct 2024 17:43:25 +0530
Subject: [PATCH] QSDK: wifi: mac80211: Fix KW issue in mac80211 driver.

Fixed KW issues which are reported in mac80211 driver
which might potentially cause crash.

Signed-off-by: Girish Kumar M P <quic_gmp@quicinc.com>
---
 net/mac80211/cfg.c        | 2 +-
 net/mac80211/mesh_plink.c | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/net/mac80211/cfg.c b/net/mac80211/cfg.c
index 29c33e4..93b56f5 100644
--- a/net/mac80211/cfg.c
+++ b/net/mac80211/cfg.c
@@ -929,7 +929,7 @@ static int ieee80211_set_monitor_channel(struct wiphy *wiphy,
 {
 	struct ieee80211_local *local = wiphy_priv(wiphy);
 	struct ieee80211_sub_if_data *sdata;
-	int ret = 0, i, def_idx = 0, hw_idx;
+	int ret = 0, i, def_idx = 0, hw_idx = 0;
 
 	if (wiphy->num_hw) {
         	hw_idx = cfg80211_get_hw_idx_by_chan(local->hw.wiphy, chandef->chan);
diff --git a/net/mac80211/mesh_plink.c b/net/mac80211/mesh_plink.c
index 0d3a481..0d88a3c 100644
--- a/net/mac80211/mesh_plink.c
+++ b/net/mac80211/mesh_plink.c
@@ -1343,6 +1343,8 @@ void mesh_rx_plink_frame(struct ieee80211_sub_if_data *sdata,
 			return;
 	}
 	elems = ieee802_11_parse_elems(baseaddr, len - baselen, true, NULL);
-	mesh_process_plink_frame(sdata, mgmt, elems, rx_status);
-	kfree(elems);
+	if (elems) {
+                mesh_process_plink_frame(sdata, mgmt, elems, rx_status);
+                kfree(elems);
+        }
 }
-- 
2.34.1

